kbs/src/plugins/implementations/splitapi/backend.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2024 by IBM Corporation
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{Context, Result};
+use std::{ffi::OsString, sync::Arc};
+use serde::Deserialize;
+
+use super::manager;
+
+
+pub const PLUGIN_NAME: &str = "splitapi";
+
+
+/// Services supported by the SplitAPI plugin
+#[async_trait::async_trait]
+pub trait SplitAPIBackend: Send + Sync {
+    /// Generate and obtain the credential for API Proxy server
+    async fn get_server_credential(&self, params: &SandboxParams) -> Result<Vec<u8>>;
+}
+
+pub struct SplitAPI {
+    pub backend: Arc<dyn SplitAPIBackend>,
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+#[serde(tag = "type")]
+pub enum SplitAPIConfig {
+    CertManager(manager::SplitAPIRepoDesc),
+}
```

Comment:
If we use `enum` here, we are hinting that potentially we could have other types of `SplitAPIConfig` besides `CertManager`. Do we? If not, I suggest to change `SplitAPIConfig` into `struct` and move members of `SplitAPIRepoDesc` directly into `SplitAPIConfig`